src/utils/formatted_timestamp.rs

```diff
@@ -0,0 +1,243 @@
+use std::error::Error as StdError;
+use std::fmt::{self, Display};
```

Comment:
```suggestion
use std::fmt;
```
Minor nit - elsewhere in the lib, we choose to impl `fmt::Display` rather than importing the trait directly.